packages/plugin-commands-installation/src/installDeps.ts

```diff
@@ -199,34 +199,43 @@ when running add/update with the --workspace option')
     }
   }
 
-  const updateMatch = opts.update && (params.length > 0) ? createMatcher(params) : null
+  let currentInput = [...params]
+
+  const ignoredPackages = (manifest.pnpm?.updateConfig?.ignoreDependencies ?? [])
+  if (opts.update && params.length === 0) {
+    currentInput = [...ignoredPackages.map(pkg => `!${pkg}`), ...currentInput]
+  }
+
+  const updateMatch = opts.update && (currentInput.length > 0) ? createMatcher(currentInput) : null
+  const hasIgnored = ignoredPackages.length > 0
   if (updateMatch != null) {
-    params = matchDependencies(updateMatch, manifest, includeDirect)
-    if (params.length === 0 && opts.depth === 0) {
+    const currentInput = matchDependencies(updateMatch, manifest, includeDirect)
+    if (currentInput.length === 0 && opts.depth === 0 && ignoredPackages.length === 0) {
       throw new PnpmError('NO_PACKAGE_IN_DEPENDENCIES',
         'None of the specified packages were found in the dependencies.')
     }
   }
 
   if (opts.update && opts.latest) {
-    if (!params || (params.length === 0)) {
-      params = updateToLatestSpecsFromManifest(manifest, includeDirect)
+    if ((!currentInput || (currentInput.length === 0)) && !hasIgnored) {
+      currentInput = updateToLatestSpecsFromManifest(manifest, includeDirect)
     } else {
-      params = createLatestSpecs(params, manifest)
+      currentInput = createLatestSpecs(currentInput, manifest)
```

Comment:
I do not have confidence that it is the right logic to run `createLatestSpecs` if some packages are specified in`Config.ignoreDependencies`, so I will rethink after

Reply:
If my understanding is correct, the dependencies which should be updated are resolved with `matchDependencies` module if some packages are specified in `Config.ignoreDependencies`, so I thought `createLatestSpecs` should be invoked when some packages are specified in `Config.ignoreDependencies`.